sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -38,9 +38,20 @@ import org.apache.spark.sql.types.StructType
 import org.apache.spark.unsafe.Platform
 import org.apache.spark.util.{NonFateSharingCache, Utils}
 
+/**
+ * Trait representing events reported from a RocksDB instance.
+ *
+ * The internal RocksDB instance can use a provider with a `RocksDBEventListener` reference to
```

Comment:
> The internal RocksDB instance can use a provider with a `RocksDBEventListener` reference

This statement is a bit confusing. Should we say just "We pass this into the internal RocksDB instance to report specific events...".

Reply:
Good idea, that makes more sense than what I had. Thanks!